functions/world.js

```diff
@@ -1,57 +1,59 @@
 const admin = require("firebase-admin");
 const functions = require("firebase-functions");
+const { HttpsError } = require("firebase-functions/lib/providers/https");
 
 const { checkAuth } = require("./src/utils/assert");
 
 const checkIsAdmin = async (uid) => {
-  await admin
-    .firestore()
-    .collection("roles")
-    .doc("admin")
-    .get()
-    .then(async (doc) => {
-      if (!doc.exists) {
-        throw new HttpsError("not-found", `'admin' doc doesn't exist.`);
-      }
-      const admins = doc.data();
-
-      if (admins.users && admins.users.includes(uid)) return;
-
-      throw new HttpsError("permission-denied", `User is not an admin`);
-    })
-    .catch((err) => {
-      throw new HttpsError(
-        "internal",
-        `Error occurred obtaining world ${worldId}: ${err.toString()}`
-      );
-    });
+  try {
+    const adminDoc = await admin
+      .firestore()
+      .collection("roles")
+      .doc("admin")
+      .get();
+
+    if (!adminDoc.exists) {
+      throw new HttpsError("not-found", `'admin' doc doesn't exist.`);
+    }
+    const admins = adminDoc.data();
+
+    if (admins.users && admins.users.includes(uid)) return;
+
+    throw new HttpsError("permission-denied", `User is not an admin`);
+  } catch (error) {
+    throw new HttpsError(
+      "internal",
+      `Error occurred obtaining world ${worldId}: ${err.toString()}`
+    );
+  }
```

Comment:
Name mismatch. Also, can probably work w/o explicit `.toString()`, assuming `HttpsError` extends `Error`

```suggestion
  } catch (error) {
    throw new HttpsError(
      "internal",
      `Error occurred obtaining world ${worldId}: ${error}`
    );
  }
```